delfin/drivers/pure/pure/pure_storage.py

```diff
@@ -0,0 +1,294 @@
+                       'healthy': constants.DiskStatus.NORMAL,
+                       'abnormal': constants.DiskStatus.ABNORMAL,
+                       'unhealthy': constants.DiskStatus.ABNORMAL,
+                       'offline': constants.DiskStatus.OFFLINE}
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.rest_handler = rest_handler.RestHandler(**kwargs)
+        self.rest_handler.login()
+
+    def list_volumes(self, context):
+        volume = []
+        volumes_return = self.rest_handler.get_all_rest_volumes()
+        if volumes_return:
+            volume = self.volume_handler(volumes_return)
+        return volume
+
+    def volume_handler(self, volumes_return):
+        volume = []
+        return_pool = self.rest_handler.get_all_pools()
+        for i in range(0, len(volumes_return)):
+            volume_dict = dict()
+            volume_object = volumes_return[i]
+            name = volume_object.get('name')
+            volume_dict['native_volume_id'] = name
+            volume_dict['name'] = name
+            total_capacity = int(int(volume_object.get('size')) / units.Ki)
+            volume_dict['total_capacity'] = total_capacity
+            used_capacity = int(int(volume_object.get('volumes')) / units.Ki)
+            volume_dict['used_capacity'] = used_capacity
+            volume_dict['free_capacity'] = total_capacity - used_capacity
+            volume_dict['storage_id'] = self.storage_id
+            volume_dict['description'] = ""
+            volume_dict['status'] = constants.StorageStatus.NORMAL
+            volume_dict['type'] = constants.VolumeType.THICK
+            volume_dict['native_storage_pool_id'] = ""
+            if return_pool:
+                for j in return_pool:
+                    volumes = j.get('volumes')
+                    if name in volumes:
+                        volume_dict['native_storage_pool_id'] = j.get('name')
+                        break
+            volume.append(volume_dict)
+        return volume
+
+    def add_trap_config(self, context, trap_config):
+        pass
+
+    def clear_alert(self, context, alert):
+        pass
+
+    def get_storage(self, context):
+        storage_object = dict()
+        result_storage = self.rest_handler.get_storage()
+        if result_storage:
+            storage_object['model'] = result_storage[0].get('hostname', "")
+            total_capacity = int(int(result_storage[0].get('provisioned',
+                                                           0)) / units.Ki)
+            storage_object['total_capacity'] = total_capacity
+            storage_object['raw_capacity'] = total_capacity
+            used_capacity = int(int(result_storage[0].get('volumes', 0)) /
+                                units.Ki)
+            storage_object['used_capacity'] = used_capacity
+            storage_object['free_capacity'] = total_capacity - used_capacity
+
+        return_storage_id = self.rest_handler.get_storage_ID()
```

Comment:
If `return_storage_id` is `None` ，None.get() will raise an exception.

Reply:
A non-null check has been performed